src/legacy/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -519,15 +518,22 @@ app.directive('dashboardApp', function ($injector) {
       updateViewMode(dashboardStateManager.getViewMode());
 
       // update root source when filters update
-      $scope.$listen(queryFilter, 'update', function () {
-        $scope.model.filters = queryFilter.getFilters();
-        dashboardStateManager.applyFilters($scope.model.query, $scope.model.filters);
+      this.updateSubscription = queryFilter.getUpdates$().subscribe({
+        next: () => {
+          $scope.model.filters = queryFilter.getFilters();
+          dashboardStateManager.applyFilters($scope.model.query, $scope.model.filters);
+        }
       });
 
       // update data when filters fire fetch event
-      $scope.$listen(queryFilter, 'fetch', $scope.refresh);
+
+      this.fetchSubscription = queryFilter.getFetches$().subscribe({
```

Comment:
Same here and in all other `.subscribe()`, maybe it can be simplified to:

```js
this.fetchSubscription = queryFilter.getFetches$().subscribe($scope.refresh);
```